File: beckend/src/Example.Application/PersonService/Service/PersonService.cs
using Example.Application.Common;
using Example.Application.PersonService.Models.Dtos;
using Example.Application.PersonService.Models.Request;
using Example.Application.PersonService.Models.Response;
using Example.Infra.Data;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace Example.Application.PersonService.Service
{
    public class PersonService : BaseService<PersonService>, IPersonService
    {
        private readonly ExampleContext _db;

        public PersonService(ILogger<PersonService> logger, ExampleContext db) : base(logger)
        {
            _db = db;
        }

        public async Task<GetAllPeopleResponse> GetAllAsync()
        {
            var entity = await _db.People.ToListAsync();
            return new GetAllPeopleResponse()
            {
                People = entity != null ? entity.Select(a => (PersonDto)a).ToList() : new List<PersonDto>()
            };
        }

        public async Task<GetByIdPersonResponse> GetByIdAsync(int id)
        {

            var response = new GetByIdPersonResponse();

            var entity = await _db.People.FirstOrDefaultAsync(item => item.Id == id);

            if (entity != null) response.Person = (PersonDto)entity;

            return response;
        }

        public async Task<CreatePersonResponse> CreateAsync(CreatePersonRequest request)
        {
            if (request == null)
                throw new ArgumentException("Request empty!");

            var newPerson = Domain.PersonAggregate.Person.Create(request.Name, request.CPF, request.IdCity, request.Age);

            _db.People.Add(newPerson);

            await _db.SaveChangesAsync();

            return new CreatePersonResponse() { Id = newPerson.Id };
        }

        public async Task<UpdatePersonResponse> UpdateAsync(int id, UpdatePersonRequest request)
        {
            if (request == null)
                throw new ArgumentException("Request empty!");

            var entity = await _db.People.FirstOrDefaultAsync(item => item.Id == id);

            if (entity != null)
            {
                entity.Update(request.Name, request.CPF, request.IdCity, request.Age);
                await _db.SaveChangesAsync();
            }

            return new UpdatePersonResponse();
        }

        public async Task<DeletePersonResponse> DeleteAsync(int id)
        {

            var entity = await _db.People.FirstOrDefaultAsync(item => item.Id == id);

            if (entity != null)
            {
                _db.Remove(entity);
                await _db.SaveChangesAsync();
            }

            return new DeletePersonResponse();
        }
    }
}

File: frontend/src/app/Services/person.service.ts
import { Injectable } from '@angular/core';
import { Person } from '../Models/person';
import { HttpHeaders } from '@angular/common/http';
import { catchError, Observable, retry } from 'rxjs';
import { BaseService } from './base.service';

@Injectable({
  providedIn: 'root'
})
export class PersonService extends BaseService {

  url: string = 'http://localhost:5194/api/Person';

  public getAll() : Observable<any> {
    return this.http
      .get<any>(this.url)
      .pipe(
        retry(2),
        catchError(this.handleError)
      );
  }

  public getById(id: number) : Observable<any> {
    return this.http
      .get<any>(this.url + '/' + id)
      .pipe(
        retry(2),
        catchError(this.handleError)
      );
  }

  public create(person: Person) {
    var data = {
      name: person.name,
      cpf: person.cpf,
      idCity: person.idCity,
      age: person.age
    };

    return this.http
     .post<string>(this.url, data)
     .pipe(
       retry(2),
       catchError(this.handleError)
     );
  }

  public update(person: Person, id: number) {
    var data = {
      name: person.name,
      cpf: person.cpf,
      idCity: person.idCity,
      age: person.age
    };

    return this.http
     .put<string>(this.url + '/' + id, data)
     .pipe(
       retry(2),
       catchError(this.handleError)
     );
  }

  public delete(id: number) {
    return this.http
      .delete<Person>(this.url + '/' + id)
      .pipe(
        retry(2),
        catchError(this.handleError)
      );
  }
}
